rfcs/0000-policy_target_scope.md

```diff
@@ -0,0 +1,267 @@
+      - type: Exact
+        name: env
+        value: production
+    backendRefs:
+    - name: bar-svc
+      port: 8080
+```
+
+It is important to consider that only _one_ rule will ever be match for any given requests. It makes for
+a simpler model to have only one _Policy_ to also only ever being matched, aligning the both behaviors.
+To apply a policy on the first rule only, the user needs to duplicate that match on the policy targeting
+`bar-route`:
+
+```yaml
+kind: PaintPolicy
+spec:
+  default:
+    color: red
+  targetRef:
+    kind: HTTPRoute
+    name: foo-route
+  rules:
+  - matches:
+    - path:
+        type: PathPrefix
+        value: /login
+```
+
+Hereby we are coloring the traffic to our `/login` on port `8081` to be `red`. That coloring _will not_
+apply to other matches. So that traffic to `bar-svc` on port `8080` would be colored `green` as per the
+`PaintPolicy` attached to `my-gw`.
+
+The resolving of _Policies_ to enforce occurs at the `HTTPRoute` or within individual `HTTPRouteRule`
+within a route. The merging strategy will resolve the individual `HTTPRouteRule` all the policies boil
+down to, stack them according to the `default` and `override` and appy the most specific one according to
+the Gateway API Policy Attachment hierarchy. Targeting a complete `HTTPRoute` is the equivalent of
+targeting each indidividual `HTTPRouteRule`s that composes the route.
+
+## What actual use-cases does this address?
+
+ - An admin restricts all traffic using a `default` at the gateway to force application developers to
+   "think" about how to authorize access to their services and have them come up with sensible `default`
+   on their `HTTPRoute`s
+ - An application developer protects its service by some rate limit, while an cluster admin could come
+   and use an `override` to mitigate a DDoS.
+
+# Reference-level explanation
+[reference-level-explanation]: #reference-level-explanation
+
+So there are a couple of things to consider here:
+
+- status reporting on "not yet" existing `HTTPRouteRule`, e.g. a Policy declares a set of `matches` that
+  don't (yet?) exist. The policy must _not_ be enforced... yet. But could eventually. This needs proper
+  reporting. 
+
+# Drawbacks
+[drawbacks]: #drawbacks
+
+- The use cases this approach doesn't address… 
+- Not very DRY... the need to keep `HTTPRouteRule`s in sync between the Policy and the `HTTPRoute` is
+  suboptimal.
+
+# Rationale and alternatives
+[rationale-and-alternatives]: #rationale-and-alternatives
+
+- [`HTTPRouteFilter`](https://gateway-api.sigs.k8s.io/references/spec/#gateway.networking.k8s.io/v1beta1.HTTPRouteFilter)
+  would be an alternative and provide finer grained control over how a policy applies, i.e. at the
+  `HTTPRouteRule` level, or even on a per `HTTPBackendRef` within an `HTTPRouteRule`. But today there is
+  no way (that I know of) for us to add a `HTTPRouteFilterType` to the list of supported
+  `HTTPRouteFilter` by a Gateway API implementor. 
```

Comment:
if there was a way to do this (for example if we proposed a GEP around third party filters), would this compliment the policy side? Does a filter at the gateway level make sense? I don't see how filters would solve the use case of wanting to set a default limit for all new endpoints